os/arch/arm/src/imxrt/imxrt_usbhost.c

```diff
@@ -0,0 +1,343 @@
+
+	imxrt_config_gpio(GPIO_USB_OTG1_OC_1);
+
+	/* Configure to receive interrupts on the overrcurrent input pin */
+#warning Missing logic
+}
+
+/***********************************************************************************
+ * Name: imxrt_usbhost_initialize
+ *
+ * Description:
+ *   Called at application startup time to initialize the USB host functionality.
+ *   This function will start a thread that will monitor for device
+ *   connection/disconnection events.
+ *
+ ***********************************************************************************/
+
+int imxrt_usbhost_initialize(void)
+{
+	pid_t pid;
+	int ret;
+
+#if 0
+	/* First, register all of the class drivers needed to support the drivers
+	 * that we care about
+	 */
+
+#ifdef CONFIG_USBHOST_HUB
+	/* Initialize USB hub support */
+
+	ret = usbhost_hub_initialize();
+	if (ret < 0) {
+		syslog(LOG_ERR, "ERROR: usbhost_hub_initialize failed: %d\n", ret);
+	}
+#endif
+
+#ifdef CONFIG_USBHOST_MSC
+	/* Register theUSB host Mass Storage Class */
+
+	ret = usbhost_msc_initialize();
+	if (ret != OK) {
+		uerr("ERROR: Failed to register the mass storage class: %d\n", ret);
+	}
+#endif
+
+#ifdef CONFIG_USBHOST_CDCACM
+	/* Register the CDC/ACM serial class */
+
+	ret = usbhost_cdcacm_initialize();
+	if (ret != OK) {
+		uerr("ERROR: Failed to register the CDC/ACM serial class\n");
+	}
+#endif
+
+#ifdef CONFIG_USBHOST_HIDKBD
+	/* Register the USB host HID keyboard class driver */
+
+	ret = usbhost_kbdinit();
+	if (ret != OK) {
+		uerr("ERROR: Failed to register the KBD class\n");
+	}
+#endif
+#endif
+
+	/* Then get an instance of the USB EHCI interface. */
+
+	g_ehciconn = imxrt_ehci_initialize(0);
+	if (!g_ehciconn) {
+		IMXLOG("ERROR: imxrt_ehci_initialize failed\n");
+		return -ENODEV;
+	}
+
+	/* Start a thread to handle device connection. */
+	pid = task_create("EHCI Monitor", CONFIG_USBHOST_DEFPRIO, CONFIG_USBHOST_STACKSIZE, (main_t) ehci_waiter, (FAR char *const *)NULL);
+	if (pid < 0) {
+		IMXLOG("ERROR: Failed to create ehci_waiter task: %d\n");
+		return -ENODEV;
+	}
+
+	return OK;
+}
+
+/***********************************************************************************
+ * Name: imxrt_usbhost_vbusdrive
+ *
+ * Description:
+ *   Enable/disable driving of VBUS 5V output.  This function must be provided by
+ *   each platform that implements the OHCI or EHCI host interface
+ *
+ * Input Parameters:
+ *   rhport - Selects root hub port to be powered host interface. Since the LPC31
+ *     has only a downstream port, zero is the only possible value for this
+ *     parameter.
+ *   enable - true: enable VBUS power; false: disable VBUS power
+ *
+ * Returned Value:
+ *   None
+ *
+ ***********************************************************************************/
+
+void imxrt_usbhost_vbusdrive(int rhport, bool enable)
+{
+	uinfo("RHPort%d: enable=%d\n", rhport + 1, enable);
+
+	/* The LPC3131 has only a single root hub port */
+
+#if 0
```

Comment:
Please remove the code instead of #if 0 if it is not required.

Reply:
I removed it